apps/balancer-tools/src/app/apr/(utils)/fetchForDateRange.ts

```diff
@@ -60,27 +61,43 @@ export async function fetchDataForDateRange({
 
   const typesArray = types?.split(",").map((t) => t.trim());
 
+  const yieldAprSum = db
+    .select({
+      poolExternalId: yieldTokenApr.poolExternalId,
+      timestamp: yieldTokenApr.timestamp,
+      valueSum: sql<number>`sum(${yieldTokenApr.value})`.as("valueSum"),
+    })
+    .from(yieldTokenApr)
+    .groupBy(yieldTokenApr.poolExternalId, yieldTokenApr.timestamp)
+    .as("yieldTokenAprSum");
+
   const poolAprForDate = db
     .select({
       poolExternalId: swapFeeApr.poolExternalId,
       avgApr:
-        sql<number>`cast(avg(coalesce(${swapFeeApr.value},0) + coalesce(${vebalApr.value},0)) as decimal)`.as(
+        sql<number>`cast(sum(coalesce(${swapFeeApr.value},0) + coalesce(${vebalApr.value},0) + coalesce(${yieldAprSum.valueSum},0)) / count(${poolSnapshots.timestamp}) as decimal)`.as(
           "avgApr",
         ),
-      avgFeeApr: sql<number>`cast(avg(${swapFeeApr.value}) as decimal)`.as(
-        "avgFeeApr",
-      ),
-      avgVebalApr: sql<number>`cast(avg(${vebalApr.value}) as decimal)`.as(
-        "avgVebalApr",
-      ),
+      avgFeeApr:
+        sql<number>`cast(sum(${swapFeeApr.value}) /  count(${poolSnapshots.timestamp})  as decimal)`.as(
+          "avgFeeApr",
+        ),
+      avgVebalApr:
+        sql<number>`cast(sum(${vebalApr.value}) /  count(${poolSnapshots.timestamp})  as decimal)`.as(
+          "avgVebalApr",
+        ),
       avgVolume:
-        sql<number>`cast(avg(${poolSnapshots.swapVolume}) as decimal)`.as(
+        sql<number>`cast(sum(${poolSnapshots.swapVolume}) /  count(${poolSnapshots.timestamp})  as decimal)`.as(
```

Comment:
This query had been repeated in the file above, no?

Reply:
yes, they are meant for two different occasion